### ISSUE TITLE
[Question] Unable to use yt-dlp on debian 9

### ISSUE THREAD

**Issue by triplesixman** [yt-dlp/yt-dlp#608]
## Checklist

- [x] I'm asking a question
- [x] I've looked through the README and FAQ for similar questions
- [x] I've searched the bugtracker for similar questions including closed ones

## Question

Following my post on the github of youtube-dl (https://github.com/ytdl-org/youtube-dl/issues/29326), I would like to switch to yt-dlp but so far it is impossible to use it.

I'm on debian 9, I ran the commands : 

```
sudo curl -L https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp -o /usr/local/bin/yt-dlp
sudo chmod a+rx /usr/local/bin/yt-dlp
```

And when I try to download any video with "yt-dlp myurl", here is the error:

> Traceback (most recent call last):
>   File "/usr/lib/python3.5/runpy.py", line 193, in _run_module_as_main
>     "__main__", mod_spec)
>   File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
>     exec(code, run_globals)
>   File "/usr/local/bin/yt-dlp/__main__.py", line 16, in <module>
>   File "<frozen importlib._bootstrap>", line 969, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
>   File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
>   File "/usr/local/bin/yt-dlp/yt_dlp/__init__.py", line 16, in <module>
>   File "<frozen importlib._bootstrap>", line 969, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
>   File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
>   File "/usr/local/bin/yt-dlp/yt_dlp/options.py", line 22, in <module>
>   File "<frozen importlib._bootstrap>", line 969, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 954, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 896, in _find_spec
>   File "<frozen importlib._bootstrap_external>", line 1147, in find_spec
>   File "<frozen importlib._bootstrap_external>", line 1123, in _get_spec
>   File "<frozen importlib._bootstrap_external>", line 1104, in _legacy_get_spec
>   File "<frozen importlib._bootstrap>", line 444, in spec_from_loader
>   File "<frozen importlib._bootstrap_external>", line 541, in spec_from_file_location
>   File "/usr/local/bin/yt-dlp/yt_dlp/cookies.py", line 44
>     KEYRING_UNAVAILABLE_REASON = f'due to unknown reasons{bug_reports_message()}'
>                                                                                 ^
> SyntaxError: invalid syntax

However, I have python2 and python3 installed:

> root@linux:~ python2 -V
> Python 2.7.13
> root@linux:~ python3 -V
> Python 3.5.3

Any idea ?

One more question, will yt-dlp be enough to avoid the youtube throttling or do we have to put "--extractor-args youtube:player_client=android --throttled-rate 100K" in parameter?

**Comment by fstirlitz** [comment]
yt-dlp requires Python ≥ 3.6: a61f4b287b6c6532b1da198df87b97a122790e34. Debian stretch is oldstable. Upgrade to buster at the very least; especially that bullseye is coming out any day now, so you’ll still be on oldstable anyway.

Though it may also work if you try monkey-patching this line to avoid f-strings.

**Comment by nyuszika7h** [comment]
Or you can install a newer Python version with [pyenv](https://github.com/pyenv/pyenv).

**Comment by pukkandan** [comment]
> Though it may also work if you try monkey-patching this line to avoid f-strings.

Probably not since we've started using fstrings everywhere

> One more question, will yt-dlp be enough to avoid the youtube throttling or do we have to put "--extractor-args youtube:player_client=android --throttled-rate 100K" in parameter?

`--throttled-rate 100K` is not needed in most cases, but is also harmless if added. Dont use the other one

**Comment by triplesixman** [comment]
> Or you can install a newer Python version with [pyenv](https://github.com/pyenv/pyenv).

Great, I managed to install python 3.9.6 thanks to pyenv. I saw that it was also necessary to insert variables in the profile file.

I have one last problem that prevents my php script to work, it's that when I execute my php script in command line it works, yt-dlp is well executed (with exec). 
But from a web browser not. In the error logs, it always displays the python 3.5.3 version. However when I do "python -V" in the terminal I get 3.9.6. I tried to reboot the server but same thing.

I think I'm missing one last place to edit to fill in the 3.9.6 but I haven't found where yet. If someone has an idea, I'm interested!
